src/app/shared/scroll/scroll.directive.ts

```diff
@@ -0,0 +1,49 @@
+import {Directive, output} from '@angular/core';
+
+@Directive({
+    selector: '[appScroll]',
+    standalone: true,
+    host: {
+        '(scroll)': 'onScroll($event)',
+    },
+})
+export class ScrollDirective {
+    // eslint-disable-next-line @typescript-eslint/no-inferrable-types
+    private scrollTopSaved: number = 0;
+    readonly scrollLoadNextData = output<Event>();
+
+    onScroll(event: Event) {
+        // eslint-disable-next-line prefer-const
+        let scrollTopNew = (event.target as HTMLElement)?.scrollTop;
+        const scrollHeight = (event.target as HTMLElement)?.scrollHeight;
+        const clientHeight = (event.target as HTMLElement)?.clientHeight;
```

Comment:
Не совсем понял идею с `?.`, какую проблему он решает? 🤔

Reply:
А вообще, рекомендую тут заиспользовать деструктуризацию, она сделает жизнь значительно проще и код будет читаемей на мой взгляд:
```suggestion
    onScroll({target}: Event) {
        const {scrollTop, scrollHeight, clientHeight} = target as HTMLElement;
```

Reply:
А почему `scrollTopNew` - через let объявлен?